### ISSUE TITLE
[rllib]PPO with action branching?

### ISSUE THREAD

**Issue by iceman126** [ray-project/ray#7194]
### PPO with action branching?
Ray version: 0.8.0
Tensorflow Version: 1.14.0
OS: Ubuntu 18.04

I'm currently working on training action branching agents with PPO. What else do I need to do besides set the action space to something like `gym.spaces.Tuple([gym.spaces.Discrete(3), gym.spaces.Discrete(5)])`, or I need to write a custom loss function? I was wondering if the gradients would be correct.

**Comment by ericl** [comment]
Yeah that's all you need for PPO. The action will be automatically computed for the space.

**Comment by iceman126** [comment]
Thank you for your answer. It helped a lot.
